tidy3d/web/api/container.py

```diff
@@ -391,16 +392,18 @@ def load_sim_data(self, task_name: str) -> SimulationDataType:
             verbose=False,
         )
 
-    def items(self) -> Tuple[TaskName, SimulationDataType]:
```

Comment:
not sure if we want to remove this. The goal of `BatchData.items()` is to load each of the `sim_data` objects one by one from file without loading them all in memory. So I think it's possible we need to overwrite the `items()` and the `values()` accordingly?

Reply:
but i thinl that‘s exactly what a mapping does? values and items are generators

Reply:
hmm ok. This might require some testing. 

so my experience originally was that just having a `batch_data : dict[str, SimulationData]` in memory was problematic. So this BatchData basically hid this by making things `dict`-like but the iteration was done by loading the values into memory one by one. I'm not sure if just making the `values()` a generator fixes this? but I could be wrong. is there a good way to test?

Reply:
this exactly how those views in mappings are implemented: https://github.com/python/cpython/blob/ef530ce7c61dc16387fb68d48aebfbbbfe02adbb/Lib/_collections_abc.py#L897-L899

essentially, `__getitem__` and `__iter__` define everything that is needed for the implementation of `.values()`, `.keys()`, and `.items()`. since both of these methods are defined in terms of single simulations, and the derived methods are generators. loading simulation data from disk is triggered by `__getitem__` for that particular item. a test for this would essentially test whether generators in python work?

the `SimulationData` will only be in memory once you have iterated over those generators and stored the results, i.e. something like `list(batch_data.values()`.